src/shared/sd-dns-resolver.c

```diff
@@ -0,0 +1,354 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
```

Comment:
hmm, implementations for sd-*.h apis are usually found in libsystemd/ or libsystemd-network/ (the latter sounds more appropriate here)

i.e. sd-*.h is for *public* apis or apis that are supposed to become public. hence they need to be implemented in libsystemd/ or libsystemd-netwwork.

Reply:
I'm looking at this again now and having a hard time moving everything into place. I can mostly shove this into libsystemd-network, but the svc param related stuff is also used by libsystemd-resolve-core which I'm guessing should not depend on libsystemd-network.

Reply:
@keszybz any suggestion on how to sort this out and where to put this code? This has been held back for quite a while, we should find an acceptable solution and move forward

Reply:
honestly, i think making resolved depend on libsystemd-network would be OK.

That said, you could also stick it into src/libsystemd/sd-network/, it seems quite fitting there.